cpp/src/utils.cpp

```diff
@@ -246,6 +246,126 @@ legate::Type to_legate_type(cudf::type_id dtype)
   }
 }
 
+legate::Type to_legate_type(arrow::Type::type code)
+{
+  switch (code) {
+    case arrow::Type::BOOL: {
+      return legate::bool_();
+    }
+    case arrow::Type::INT8: {
+      return legate::int8();
+    }
+    case arrow::Type::INT16: {
+      return legate::int16();
+    }
+    case arrow::Type::INT32: {
+      return legate::int32();
+    }
+    case arrow::Type::INT64: {
+      return legate::int64();
+    }
+    case arrow::Type::UINT8: {
+      return legate::uint8();
+    }
+    case arrow::Type::UINT16: {
+      return legate::uint16();
+    }
+    case arrow::Type::UINT32: {
+      return legate::uint32();
+    }
+    case arrow::Type::UINT64: {
+      return legate::uint64();
+    }
+    case arrow::Type::FLOAT: {
+      return legate::float32();
+    }
+    case arrow::Type::DOUBLE: {
+      return legate::float64();
+    }
+    case arrow::Type::STRING: {
+      return legate::string_type();
+    }
+    default: throw std::invalid_argument("unsupported Arrow datatype");
+  }
+}
+
+std::shared_ptr<arrow::DataType> to_arrow_type(cudf::type_id code)
```

Comment:
This looks fine.  Do we actually ever use it without having a `cudf::data_type`, though?

I think I would slightly prefer to just pass `const cudf::data_type &cudf_type`, since I suspect there will be (small/odd) places where the type code may not be enough and I doubt there are many places where we only have the ID.

Reply:
Will overhaul the type conversions in a later PR.